es/es-repository-s3/src/main/java/org/elasticsearch/repositories/s3/S3ClientSettings.java

```diff
@@ -163,7 +163,7 @@ private S3ClientSettings(AWSCredentials credentials, String endpoint, Protocol p
         return Map.of(DEFAULT, getClientSettings(settings));
     }
 
-    static boolean checkDeprecatedCredentials(Settings repositorySettings) {
+    static boolean checkRepositoryCredentials(Settings repositorySettings) {
```

Comment:
Could this also already be removed altogether (given the mandatory settings logic) ?